File: src/leetcode/l_257.go
package main

import (
	"fmt"
	"strconv"
)

type treeNode struct {
	left  *treeNode
	right *treeNode
	val   int
}

func create() *treeNode {
	num := -1
	fmt.Scanf("%d", &num)
	if num == -1 {
		return nil
	}
	node := treeNode{}
	node.val = num
	node.left = create()
	node.right = create()
	return &node
}

var res []string = []string{}

var tmp string = ""

func travsel(node *treeNode) []string {
	if node == nil {
		return res
	}
	//list = append(list, node.Val)
	tmp += strconv.Itoa(node.val) + "->"
	if node.left == nil && node.right == nil {
		rs := []rune(tmp)
		temp := make([]rune, len(rs)-2)
		s := string(temp)
		//tmp = tmp[0 : len(tmp)-2]
		fmt.Println(s)
		res = append(res, s)
	}
	travsel(node.left)
	travsel(node.right)
	if len(tmp) > 3 {
		tmp = tmp[0 : len(tmp)-3]
	}
	return res
}

func main() {
	root := create()
	fmt.Println(root)
	strings := travsel(root)
	fmt.Println(strings)
}

File: src/leetcode/l_257.java
package leetcode;

import java.util.Arrays;
import java.util.LinkedList;
import java.util.List;

public class l_257 {
    public static void main(String[] args) {
        TreeNode treeNode = TreeNode.create();
        System.out.println(Arrays.toString(binaryTreePaths(treeNode).toArray()));

    }

    private static LinkedList<Integer> list = new LinkedList<>();
    private static List<String> res = new LinkedList<>();

    public static List<String> binaryTreePaths(TreeNode root) {
        if (root == null) {
            return res;
        }
        list.add(root.val);
        if (root.left == null && root.right == null) {
            System.out.println(Arrays.toString(list.toArray()));
            StringBuilder tmp = new StringBuilder();
            for (int i = 0; i < list.size() - 1; i++) {
                tmp.append(list.get(i)).append("->");
            }
            tmp.append(list.get(list.size() - 1));
            res.add(tmp.toString());
        }
        binaryTreePaths(root.left);
        binaryTreePaths(root.right);
        list.removeLast();
        return res;
    }
}
